app/grants/views.py

```diff
@@ -570,80 +590,118 @@ def get_grants(request):
         _grants = _grants.filter(vector_column=query)
 
     if not idle_grants:
+        # 7. Filter grants which are stale (last update was > 3 months )
         _grants = _grants.filter(last_update__gt=three_months_ago)
 
     if state == 'active':
+        # 8. Filter grants which are active
         _grants = _grants.active()
 
-    if grant_type != 'all' and grant_type != 'me' and grant_type != 'collections':
-        _grants = _grants.filter(grant_type__name=grant_type)
+    if grant_types and grant_types not in ['all', 'collections']:
+        # 9. Fetch grants which have mentioned grant_types
+        types= grant_types.split(',')
+        type_query= Q()
+        for grant_type in types:
+            type_query |= Q(grant_type__name=grant_type)
+
+        _grants = _grants.filter(type_query)
 
     if following and request.user.is_authenticated:
+        # 10. Filter grants having matching title & description
         favorite_grants = Favorite.grants().filter(user=request.user).values('grant_id')
         _grants = _grants.filter(id__in=Subquery(favorite_grants))
 
-    clr_prediction_curve_schema_map = {10**x: x+1 for x in range(0, 5)}
-    if sort_by_clr_pledge_matching_amount in clr_prediction_curve_schema_map.keys():
-        sort_by_index = clr_prediction_curve_schema_map.get(sort_by_clr_pledge_matching_amount, 0)
-        field_name = f'clr_prediction_curve__{sort_by_index}__2'
-        _grants = _grants.order_by(f"-{field_name}")
+    if grant_tags:
+        tags = grant_tags.split(',')
+        # 11. Fetch grants which have mentioned grant_tags
+        tag_query= Q()
+        for tag in tags:
+            tag_query |= Q(tags__name=tag)
+
+        _grants = _grants.filter(tag_query)
+
+    if tenants:
+        # 12. Fetch grants which have mentioned tenants
+        tenant_query = Q()
+        for tenant in tenants.split(','):
+            if tenant in tenants:
+                tenant = tenant.lower()
+                tenant_filter = tenant + '_payout_address' if tenant != 'eth' else 'admin_address'
+                tenant_query = ~Q(('%s' % tenant_filter, '0x0'))
+
+            # _grants = _grants.exclude(Q(**{tenant_filter: '0x0'}))
+        _grants = _grants.filter(tenant_query)
+
+
+    # 13. Sort filtered grants
+    if sort:
+        if 'match_pledge_amount_' in sort:
+            sort_by_clr_pledge_matching_amount = int(sort.split('amount_')[1])
+            clr_prediction_curve_schema_map = {10**x: x+1 for x in range(0, 5)}
+            if sort_by_clr_pledge_matching_amount in clr_prediction_curve_schema_map.keys():
+                sort_by_index = clr_prediction_curve_schema_map.get(sort_by_clr_pledge_matching_amount, 0)
+                field_name = f'clr_prediction_curve__{sort_by_index}__2'
+                _grants = _grants.order_by(f"-{field_name}")
+        elif sort in ['-amount_received_in_round', '-clr_prediction_curve__0__1']:
+            grant_type_obj = GrantType.objects.filter(name=grant_type).first()
```

Comment:
@thelostone-mc seems this grant_type is not defined when filtering by clr_prediction_curve, you can check it now using the latest.

Reply:
hmm this filter is weird ! It looks like we simple apply a filter on top of it instead of sorting. I'm commenting this piece out to keep it simple